app/hooks/useJsonSchemaDiff.ts

```diff
@@ -0,0 +1,92 @@
+import diff from 'deep-diff';
+
+const addObjectToDiffMap = (path, lhs, rhs, idDiffMap) => {
+  if (typeof lhs !== 'object' && typeof rhs !== 'object') {
+    idDiffMap[path] = {lhs: lhs ?? null, rhs: rhs ?? null};
+    return;
+  }
+  if (typeof lhs === 'object' || typeof rhs === 'object') {
+    const keys = new Set<string>();
+
+    if (lhs) Object.keys(lhs).forEach((k) => keys.add(k));
+    if (rhs) Object.keys(rhs).forEach((k) => keys.add(k));
+
+    keys.forEach((k) =>
+      addObjectToDiffMap(`${path}_${k}`, lhs?.[k], rhs?.[k], idDiffMap)
+    );
+  }
+};
```

Comment:
Just a thought - I wonder if it would be useful to add a "max depth" parameter to avoid crash with circular references?

Reply:
Since the data comes from a JSON object, we can't have any circular reference in practice